composer/utils/inference.py

```diff
@@ -57,6 +57,29 @@ class ExportFormat(StringEnum):
     ONNX = 'onnx'
 
 
+def _move_sample_input_to_device(sample_input: Union[torch.Tensor, dict, list, Tuple],
```

Comment:
we do have `map_collection` if that helps (https://github.com/mosaicml/composer/blob/dev/composer/utils/iter_helpers.py#L13) here

Reply:
I think the recursion in this function actually matters, which makes it a bit trickier to use the helper, but definitely useful to know about.